thirdparty/github.go

```diff
@@ -1765,6 +1765,104 @@ func GetMergeablePullRequest(ctx context.Context, issue int, githubToken, owner,
 	return pr, nil
 }
```

Comment:
Starting a random thread for this so we can add comments in one place:
>@ablack12 @malikchaya2 @minnakt I added you as reviewers, since you've all typed merge queue code or done a code review, and I'm not sure what the right approach for migration should be. I'm curious if you have thoughts about the right approach.

I think what you've said about going the route of (3) is reasonable given the risk. Maybe we let users know (with a warning message or something, Minna would know better) if they toggle the merge queue on that it won't be effective without the Evergreen branch protection? Or maybe best scenario, we don't allow them to turn it on if they don't have the check in place.

Reply:
Oh I didn't consider a hybrid approach. We could check to see if they have a branch protection rule, and then if they don't, we could offer to try to set it for them. This would mean that they'd be explicitly opting in to letting Evergreen mess with their settings.

Reply:
I like that approach!